### ISSUE TITLE
import/order with named fails when exporting uses spread operator.

### ISSUE THREAD

**Issue by UnderKoen** [import-js/eslint-plugin-import#3145]
Rule:
```js
'import/order': [
  'warn',
  {
    named: true,
  },
],
```

minimal repro case:
```js
const test = {
  a: 1,
  browser: 2,
};

module.exports = {
  ...test,
  target: 'node',
};
```

**Comment by ljharb** [comment]
What does native node do when importing this module into ESM? I assume it breaks cjs-module-lexer's static analysis as well?

**Comment by UnderKoen** [comment]
@ljharb It doesn't break but it ignores it.

https://www.npmjs.com/package/cjs-module-lexer#:~:text=Object%20properties%20that%20are%20not%20identifiers%20or%20string%20expressions%20will%20bail%20out%20of%20the%20object%20detection%2C%20while%20spreads%20are%20ignored%3A

**Comment by ljharb** [comment]
By ignore, that means it won’t treat those as named imports, which would match the current behavior of this plugin?

**Comment by UnderKoen** [comment]
Currently it crashes. As `node.right.properties[i].key` is undefined. That why there is just an truthy check added in #3146

**Comment by ljharb** [comment]
aha :-) ok, i'll give that PR a review then.
